docs/migration.rst

```diff
@@ -4,34 +4,92 @@ Migrating from catkin_make
-  This means there will be no warnings for unused variables passed to ``cmake``.
+- All targets in each of a package's dependencies are guaranteed to have been built before the current package.
+- Packages do not need to define target dependencies on ROS messages built in other packages.
+- It passes the same CMake command line arguments to multiple packages.
+- Plain CMake packages can be built if they each have a ``package.xml`` file with the appropriate `<build_type> tag <http://www.ros.org/reps/rep-0140.html#build-type>`_.
 
-IDE Integration
-^^^^^^^^^^^^^^^
+Additional Differences with ``catkin build``
+--------------------------------------------
 
-Since all packages are built in isolation with ``catkin build``, you can't rely on CMake's IDE integration to generate a single project for your entire workspace.
+In addition to the differences due to isolation, ``catkin build`` is also different from ``catkin_make_isolated`` in the following ways:
+
+- It builds packages in parallel, using an internal job server to distribute load.
+- It puts products into hidden directories, and then symbolically links them into the **devel space** (by default).
+- It stores persistent configuration options in a ``.catkin_tools`` directory at the root of your workspace.
+- It passes ``--no-warn-unused-cli`` to the ``cmake`` command since not all packages accept the same CMake arguments.
+- It generates ``.catkin`` files where each source package is listed, individually, instead of just listing the source space for the workspace.
+  This leads to similar ``ROS_PACKAGE_PATH`` variables which list each package source space.
+
+Step-by-Step Migration
+^^^^^^^^^^^^^^^^^^^^^^
+
+Most problems users will encounter when migrating from ``catkin_make`` to ``catkin build`` are due to hidden bugs in packages which previously relied on side-effects from their dependencies to build.
+The best way to debug these problems before switching to the entirely new tool, is to use ``catkin_make_isolated`` first.
+Note that all three of these tools can share **source spaces**, but they must use their own build, devel, and install spaces.
+
+1. Verify that your packages already build with ``catkin_make``:
+----------------------------------------------------------------
+
+To make iterating easier, use ``catkin_make`` with build and devel spaces with the suffix ``_cm`` so that they do not collide with the other build tools:
+
+.. code-block:: bash
+
+    cd /path/to/ws
+    catkin_make --build build_cm --devel devel_cm --cmake-args [CMAKE_ARGS...] --make-args [MAKE_ARGS...]
+
+If your packages build and other appropriate tests pass, continue to the next step.
+
+2. Verify that your packages build in isolation:
+------------------------------------------------
+
+Use ``catkin_make_isolated`` with build and devel spaces with the suffix ``_cmi``, and make sure your packages build in isolation.
```

Comment:
But I guess since you mention a custom suffix for every command it is more consistent that way. So forget about this.